spec/support/link_check_matchers.rb

```diff
@@ -10,4 +10,14 @@
       raise "Failed when checking URL: \"#{check_link}\""
     end
   end
+end
+
+RSpec::Matchers.define :be_a_working_link do
+  match do |check_link|
+    begin
+      good_statuses.include? Faraday.head(check_link).status
+    rescue => e
+      raise "Failed when checking \"#{check_link}\": #{e.message}"
```

Comment:
to my knowledge, it does not throw any of the stack trace away. `e.message` is not part of the trace, just the error message that gets printed first. All this does is prepend the failing URL onto the message, which Faraday does not print.